### ISSUE TITLE
Visualize upload states

### ISSUE THREAD

**Issue by johnjohndoe** [sensebox/senseBox-bike-Arduino#8]
# Context
- Uploading data happens when the device is connected to a known WiFi.
- As a user I **do not know** when exactly the uploads happens, if at all and when it is done.

# Proposal
- The **device should signal** via a documented color code that the WiFi connection is established and that the upload
  - is starting
  - is happening
  - is done
- The upload states could be visualized as specific colors or via a blinking pattern.

**Comment by goetzvr** [comment]
When the upload is done, the box shuts down.  So this is, at least for me, a good marker that the upload is done.

**Comment by johnjohndoe** [comment]
How do you know that the upload actually happened? How do you know that the box has shut down (light off)?

**Comment by goetzvr** [comment]
During the workshop they explained: If you leave the box idling on your desk it will search for the configured WiFI. When it is connected it will upload the collected data and then shut down or go to a [deep sleep](https://github.com/sensebox/senseBox-bike-Arduino/blob/ba2977e16af43f080c803c904fdd1dc789f8fbd9/futuriumSDNoAp.ino#L161).
You have to (hard) turn it off and on  to restart the box.

**Comment by johnjohndoe** [comment]
True. I remember the explanation, too. Thank you for the code pointer.

I however would find it useful if I could recognize just by observing the device if an upload happens. Specially, the first time one becomes somewhat uncertain if it works. The WiFi credentials might have a typo. The only way to find out if the upload worked is to check the website. Thus, I would welcome a visual status feedback of the device a UX improvement.

**Comment by Thiemann96** [comment]
I thought about this too but one thing kept me from actually implementing this.

The powerbank which powers the MCU demands a "minimum energy output". When this minimum energy output is not met the powerbank shuts off(Abschaltautomatik)) This way for example the powerbank would not overcharge your phone if it is at 100%. The senseBox only produces this minimum energy output with all sensors on and connected. So in order to give visual feedback that the upload was successful the box and all its sensors would need to be powered. 

Now, one could implement a solution which gives visual feedback and then turn the box off after some minutes but not powering anything but the LED would not work here. In the end I decided instead of all the hassle that would come with implementing this, the visual feedback that the box turns off would be more straight forward and easier. I see your point though that some kind of feedback would be nice for the upload states.

**Comment by Thiemann96** [comment]
How do you feel about a solution that lets the LED blinks for 5 minutes (according to the different color codes you mentioned) and then shuts itself off ?

**Comment by johnjohndoe** [comment]
Thank you for the explanation regarding the power bank restrictions.

> [...] lets the LED blink [...]

... to visualize that the upload is taking place. I think this would already be an improvement. :heart:
